scripts/devel/build_and_install.py

```diff
@@ -77,7 +117,21 @@ def main(argv=sys.argv[1:]):
         if args.clean_after:
             clean_workspace(args.workspace_root)
 
-    return rc
+    if not args.run_abichecker:
+        return rc
+
+    with Scope('SUBSECTION', 'use abi checker'):
+        rc = call_abi_checker(
+            [args.workspace_root],
+            args.ros_version,
+            env)
+    # Never fail a build because of abi errors but make them
+    # unstable by printing MAKE_BUILD_UNSTABLE. Jenkins will
+    # use a`plugin to make it
+    if rc != 0:
+        print('MAKE_BUILD_UNSTABLE')
+
+    return 0
```

Comment:
Are we intentionally suppressing the ABI checker return code by not returning `rc` here?
Should we return the ABI checker code, or even perhaps the build command code?

Reply:
> Are we intentionally suppressing the ABI checker return code by not returning `rc` here?

Yes. The approach in this pull request is that abi-checker can return unstable but never fails the build. 

>  Should we return the ABI checker code, or even perhaps the build command code?

My impression is that if we do it the build will fail with `build_and_install.py` returning non zero code.

Reply:
In that case, should the ABI checker run if the build command rc != 0?
Because right now it appears to run without checking the build command rc at all.

Reply:
that was indeed a bug, good catch f0161fe7bc4b4535e1844dfe1910c7048303af59